### ISSUE TITLE
remove pin on Twisted<=16.1.1, all versions (including current 16.2.0) are ok

### ISSUE THREAD

**Issue by warner** [crossbario/autobahn-python#680]
A couple of weeks ago, in reponse to #641, setup.py acquired an `extras_require[twisted]` that pins the Twisted version:

```
    "Twisted>=15.5,<=16.1.1"        # MIT license
```

At the time, Twisted-16.1.1 was current, and Twisted trunk had committed a change which broke autobahn pretty badly (https://twistedmatrix.com/trac/ticket/8191 , landed on 29-Mar-2016 in twisted git revision 4f9e36d8ad21f9d73369f4d506bab30f61c2ad71). We were worried that the next version (16.1.2? 16.2.0?) would include the change, breaking the combination of (most-recent-Twisted-release) and (most-recent-Autobahn-release). So the pin froze the acceptable version of Twisted to the current release at that time (16.1.1), in anticipation of a broken release.

On 23-Apr-2016, Twisted reverted that change, in commit d03fee6d464a651e27f86836b15444d5b47bc734. The breaking change never made it into a release, and Twisted-16.2.0 was released on 17-May-2016 with no problems. All versions of Twisted (16.1.0, 16.1.1, 16.2.0) work with autobahn (at least w.r.t. the #641 web.Resource issue).

I'm wondering if autobahn could remove the pin. I have a project that needs both Autobahn and Twisted, and since pip doesn't have a full dependency resolver, running my `pip install magic-wormhole` gets me Twisted-16.2.0 and autobahn-0.14.0, and then my commands fail because the installed versions don't meet the pin requirements (under setuptools-21.2.1):

```
(ve4) % wormhole --version
Traceback (most recent call last):
 ...
"ve4/lib/python2.7/site-packages/pkg_resources/__init__.py", line 834, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.ContextualVersionConflict: (Twisted 16.2.0 (/Users/warner/stuff/tahoe/magic-wormhole/ve4/lib/python2.7/site-packages), Requirement.parse('Twisted<=16.1.1,>=15.5; extra == "twisted"'), set(['autobahn']))
```

To allow my users to run `pip install magic-wormhole` successfully, I'd need to pin twisted back myself. I'll do that if necessary (I'm trying to make a 1.0 release in time for my pycon presentation, and I want it to be installable by then), but it will make downstream packaging more difficult, and I think it'd be cleaner all around if the autobahn `setup.py` removed the constraint, now that it turned out to be unnecessary.

thanks!
 -Brian

**Comment by meejah** [comment]
I can confirm that Twisted 16.2.0 passes all tests locally; #681 changes the Twisted dep to ``>=16.2.0``

**Comment by warner** [comment]
I see this was fixed in #682 and made it into today's 0.14.1 release. Thanks!
